executor/metrics_reader.go

```diff
@@ -76,14 +75,7 @@ func (e *MetricRetriever) retrieve(ctx context.Context, sctx sessionctx.Context)
 	}
 	for _, quantile := range quantiles {
 		var queryValue pmodel.Value
-		// Add retry to avoid network error.
-		for i := 0; i < 10; i++ {
-			queryValue, err = e.queryMetric(ctx, sctx, queryRange, quantile)
-			if err == nil || strings.Contains(err.Error(), "parse error") {
-				break
-			}
-			time.Sleep(100 * time.Millisecond)
-		}
+		queryValue, err = e.queryMetric(ctx, sctx, queryRange, quantile)
```

Comment:
I think we still need the retry logic, if not retry here, should retry in `queryMetric `.